src/main/kotlin/org/rust/stdext/BitFlagsBuilder.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.stdext
+
+/**
+ * A simple utility for bitmask creation. Use it like this:
+ *
+ * ```
+ * object Foo: BitFlagsBuilder(Limit.BYTE) {
+ *     val BIT_MASK_0 = nextBitMask() // Equivalent to `1 shl 0`
+ *     val BIT_MASK_1 = nextBitMask() // Equivalent to `1 shl 1`
+ *     val BIT_MASK_2 = nextBitMask() // Equivalent to `1 shl 2`
+ *     // ...etc
+ * }
+ * ```
+ */
+abstract class BitFlagsBuilder private constructor(private val limit: Limit, startFromBit: Int) {
+    protected constructor(limit: Limit) : this(limit, 0)
+    protected constructor(prevBuilder: BitFlagsBuilder, limit: Limit) : this(limit, prevBuilder.counter)
```

Comment:
Since we use `Byte` almost anywhere, maybe set it as default parameter

```suggestion
    protected constructor(limit: Limit = Limit.Byte) : this(limit, 0)
    protected constructor(prevBuilder: BitFlagsBuilder, limit: Limit = Limit.Byte) : this(limit, prevBuilder.counter)
```

Reply:
I'd want it to be more explicit :thinking: